doc/src/dev_intro.rst

```diff
@@ -0,0 +1,152 @@
+MIGraphX Fundamentals
+======================
+
+MIGraphX provides an optimized execution engine for deep learning neural networks.
+We will cover some simple operations in the MIGraphX framework here.
+We created this documentation for contributors to the MIGraphX codebase.
+This documentation is not intended for users of the MIGraphX library.
```

Comment:
I don't think we should say this. Instead, I would suggest to point those looking to quickly get started using MIGraphX to the examples: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/tree/develop/examples/migraphx